emperor/flat_plots.py

```diff
@@ -0,0 +1,346 @@
+from matplotlib import use
+use('Agg')  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+def plot_3x3(ordination, offset, mapping, field, colors, null_kw=None,
```

Comment:
Can we make this such that an arbitrary number of dimensions is plotted? i.e. pc1 vs pc2 alone? For example, an interface like this would be more flexible:

```python

def daniels_great_plotting_function(ordination, mapping, colors, field, cols, offset, null_kw=None, focus_kw=None):
     # ...
```

The main differences are the order, and the addition of a `cols` argument that lets you specify which dimensions are to be plotted, for example if you were to pass `['pc1', 'pc2']` as the value of `cols`, then you would only get as a result  a single scatter plot showing you pc1 vs pc2.

Additionally, it would be great if the colors could be determined from the current color list (as noted by the matplotlib API), this way there's no need to create a specific map between the categories and a color, though having the ability to do so is certainly useful. In this way, I think we want to mimic seaborn's API.

Together these changes would get rid of the `offset` argument and simplify the example in your notebook.

Lastly, have you thought about having shapes be used in these plots? That's a nice thing we could leverage from matplotlib.